dip-template/runtimes/dip-provider/src/dip.rs

```diff
@@ -27,22 +27,59 @@ use runtime_common::dip::{
 use scale_info::TypeInfo;
 use sp_std::vec::Vec;
 
-use crate::{AccountId, DidIdentifier, Hash, Runtime, RuntimeEvent};
-
-#[derive(Encode, Decode, TypeInfo)]
-pub struct RuntimeApiDipProofRequest {
-	pub(crate) identifier: DidIdentifier,
-	pub(crate) version: IdentityCommitmentVersion,
-	pub(crate) keys: Vec<KeyIdOf<Runtime>>,
-	pub(crate) accounts: Vec<LinkableAccountId>,
-	pub(crate) should_include_web3_name: bool,
+use crate::{AccountId, Balances, DidIdentifier, Hash, Runtime, RuntimeEvent};
+
+pub mod runtime_api {
+	use super::*;
+
+	#[derive(Encode, Decode, TypeInfo)]
+	pub struct DipProofRequest {
+		pub(crate) identifier: DidIdentifier,
+		pub(crate) version: IdentityCommitmentVersion,
+		pub(crate) keys: Vec<KeyIdOf<Runtime>>,
+		pub(crate) accounts: Vec<LinkableAccountId>,
+		pub(crate) should_include_web3_name: bool,
+	}
+
+	#[derive(Encode, Decode, TypeInfo)]
+	pub enum DipProofError {
+		IdentityNotFound,
+		IdentityProviderError(<LinkedDidInfoProviderOf<Runtime> as IdentityProvider<DidIdentifier>>::Error),
+		MerkleProofError(DidMerkleProofError),
+	}
+}
+
+pub mod deposit {
+	use crate::{Balance, RuntimeHoldReason, UNIT};
+
+	use super::*;
+
+	use frame_support::traits::Get;
+	use pallet_deposit_storage::{StorageDepositCollectorViaDepositsPallet, MAX_NAMESPACE_LENGTH};
+	use sp_core::{ConstU128, ConstU32};
+	use sp_runtime::BoundedVec;
+
+	pub struct Namespace;
+
+	impl Get<BoundedVec<u8, ConstU32<MAX_NAMESPACE_LENGTH>>> for Namespace {
+		fn get() -> BoundedVec<u8, ConstU32<MAX_NAMESPACE_LENGTH>> {
+			(*b"DipProvider").to_vec().try_into().expect("Should never fail.")
+		}
+	}
+
+	pub const DEPOSIT_AMOUNT: Balance = 100 * UNIT;
```

Comment:
This will be a constant specified somewhere in the runtime config.

Reply:
If we want to reuse this pallet, we should put some more thought in how to configure it for different actions that require deposits.

Reply:
As discussed on Slack, the initial scope of this pallet is to support storing/removing deposits. More complex logic COULD be added in the future, if we see fit. For now I'm marking this conversation as resolved.